pandas/core/algorithms.py

```diff
@@ -431,6 +432,15 @@ def isin(comps: AnyArrayLike, values: AnyArrayLike) -> np.ndarray:
         return cast("Categorical", comps).isin(values)
 
     comps, dtype = _ensure_data(comps)
+
+    if is_numeric_dtype(dtype):
+        # Try downcasting values if comps is numeric to prevent precision
+        # loss resulting from casting values to comps its exact dtype
+        try:
```

Comment:
this is what ensure_data does, and we really don't want to use try/excepts in this routine, everything should be via is_*